src/input_arguments_handling.jl

```diff
@@ -0,0 +1,122 @@
+    n = FixedPointDecimals.max_exp10(T)
+    return f >= 0 && (n < 0 || f <= n)
+end
+
+# Separate out the types that are not pre-compiled by the parser by default
+# and return them as a single Tuple of unique types which can be passed to
+# populate_result_buffer! to trigger recompilation. See `parsecustom!`
+# in src/row_parsing.jl for how this is used.
+function _custom_types(schema::Vector{DataType})
+    # We sort the unique types to always produce the same Tuple for the same
+    # schema. But maybe the default ordering from the IdDict is good enough?
+    custom_types = sort!(collect(keys(
+            IdDict{Type,Nothing}(
+                T => nothing for T in schema if isnothing(get(Enums._MAPPING, T, nothing))
+            ))),
+        by=objectid
+    )
+    return Tuple{custom_types...}
+end
+
+# Parsers.Options constructor with our defaults
+function _create_options(;
+    delim::Union{UInt8,Char,String,Nothing}=',',
+    openquotechar::Union{UInt8,Char}='"',
+    closequotechar::Union{UInt8,Char}='"',
+    escapechar::Union{UInt8,Char}='"',
+    sentinel::Union{Missing,Nothing,Vector{String}}=missing,
+    groupmark::Union{Char,UInt8,Nothing}=nothing,
+    stripwhitespace::Bool=false,
+    truestrings::Union{Nothing,Vector{String}}=["true", "True", "TRUE", "1", "t", "T"],
+    falsestrings::Union{Nothing,Vector{String}}=["false", "False", "FALSE", "0", "f", "F"],
+    dateformat::Union{String, Dates.DateFormat, Nothing, AbstractDict}=nothing,
+    ignorerepeated::Bool=false,
+    quoted::Bool=true,
+    decimal::Union{Char,UInt8}='.',
+    ignoreemptyrows::Bool=true,
+    rounding::Union{Nothing,RoundingMode}=RoundNearest,
+)
+    return Parsers.Options(
+        sentinel=sentinel,
+        wh1=delim ==  ' ' ? '\v' : ' ',
+        wh2=delim == '\t' ? '\v' : '\t',
+        openquotechar=UInt8(openquotechar),
+        closequotechar=UInt8(closequotechar),
+        escapechar=UInt8(escapechar),
+        delim=delim,
+        quoted=quoted,
+        stripwhitespace=stripwhitespace,
+        trues=truestrings,
+        falses=falsestrings,
+        groupmark=groupmark,
+        dateformat=dateformat,
+        ignorerepeated=ignorerepeated,
+        decimal=UInt8(decimal),
+        ignoreemptylines=ignoreemptyrows,
+        rounding=rounding,
+    )
+end
+
+# Check that the schema and header are compatible
+_validate(header::Vector, schema::Vector, validate_type_map) = length(header) == length(schema) || throw(ArgumentError("Provided header and schema lengths don't match. Header has $(length(header)) columns, schema has $(length(schema))."))
+_validate(header::Vector, schema::Dict{Symbol}, validate_type_map) = !validate_type_map || issubset(keys(schema), header) || throw(ArgumentError("Provided header and schema names don't match. In schema, not in header: $(collect(setdiff(keys(schema), header))). In header, not in schema: $(setdiff(header, keys(schema)))"))
```

Comment:
i find the double negative hard to follow

is this
```suggestion
function _validate(header::Vector, schema::Dict{Symbol}, validate_type_map) 
if validate_type_map && !(issubset(keys(schema), header))
    throw(ArgumentError("Provided header and schema names don't match. In schema, not in header: $(collect(setdiff(keys(schema), header))). In header, not in schema: $(setdiff(header, keys(schema)))"))
end
```

Reply:
What double negative?